pkg/resources/management.cattle.io/v3/fleetworkspace/validator.go

```diff
@@ -84,26 +88,31 @@ func (m *Mutator) Admit(request *admission.Request) (*admissionv1.AdmissionRespo
 
 	namespace := v1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: fw.Name,
+			Name:   fw.Name,
+			Labels: map[string]string{fleetManagedLabel: "true"},
 		},
 	}
 	ns, err := m.namespaces.Create(&namespace)
 	if err != nil {
 		if !errors.IsAlreadyExists(err) {
 			return nil, err
 		}
-		// check if user has enough privilege to create fleet admin rolebinding in the namespace
-		cr, err := m.clusterroles.Cache().Get(fleetAdminRole)
+		ns, err := m.namespaces.Get(fw.Name, metav1.GetOptions{})
 		if err != nil {
 			return nil, err
 		}
-		response := &admissionv1.AdmissionResponse{}
-		auth.SetEscalationResponse(response, auth.ConfirmNoEscalation(request, cr.Rules, namespace.Name, m.resolver))
-
-		ns, err = m.namespaces.Get(namespace.Name, metav1.GetOptions{})
-		if err != nil {
-			return nil, err
+		if ns.Labels[fleetManagedLabel] != "true" {
+			return &admissionv1.AdmissionResponse{
+				Result: &metav1.Status{
+					Status:  "Failure",
+					Message: fmt.Sprintf("namespace '%s' already exists", fw.Name),
+					Reason:  metav1.StatusReasonInvalid,
+					Code:    http.StatusBadRequest,
+				},
+				Allowed: false,
+			}, nil
```

Comment:
use the admission helper function for consistent webhook messages.
```suggestion
			return admission.ResponseBadRequest(fmt.Sprintf("namespace '%s' already exists", fw.Name)), nil
```

Reply:
done in https://github.com/rancher/webhook/pull/260/commits/24334874f8922d5187275b7fdf0f1c85dfcacf5f